x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/join/JoinConfig.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.plan.logical.join;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.xpack.esql.io.stream.PlanStreamInput;
+import org.elasticsearch.xpack.esql.io.stream.PlanStreamOutput;
+import org.elasticsearch.xpack.ql.expression.NamedExpression;
+
+import java.io.IOException;
+import java.util.List;
+
+public record JoinConfig(JoinType type, List<NamedExpression> unionFields) implements Writeable {
```

Comment:
I think we could align the naming with Enrich and Lookup:
```suggestion
public record JoinConfig(JoinType type, List<NamedExpression> matchFields) implements Writeable {
```
Applies also to HashJoinExec.

In the context of joins, "union" reminds me of, well, relational unions, which are a different concept.